CMakeLists.txt

```diff
@@ -53,17 +53,24 @@ include(GNUInstallDirs)
 include(Filter)
 
 #================================================ Precompiler flags
+#LUA CONSOLE
+add_definitions(-DNO_FONTGENESIS)
 
 
 #================================================ Compiler flags
-set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Wall -Wno-int-in-bool-context")
-set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Wno-format -Wno-unused-but-set-variable")
-set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Wno-unused-variable -Wno-write-strings")
-set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Wno-sign-compare -Wno-strict-aliasing")
-set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Wno-unused-value")
-set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0")
-set(CMAKE_CXX_FLAGS_DEBUG  "${CMAKE_CXX_FLAGS_DEBUG} -O0")
-add_definitions(-fpermissive)
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-unused-variable -std=c++11 -O3 -DNDEBUG")
```

Comment:
The two lines:

```cmake
set(CMAKE_CXX_STANDARD 11)
set(CMAKE_CXX_EXTENSIONS OFF)
```

below should take care of the proper C++11 flags for each compiler automatically. That is, `-std=c++11` should not be needed here. Try the verbose build (`./configure.sh -DCMAKE_VERBOSE_MAKEFILE=1`) to see if  `-std=c++11` is included twice, and if it is, remove it.

The same goes for `-O3 -DNDEBUG`. If you do a _release build_ (`./configure Release` or just the default `./configure`), `-O3 -DNDEBUG` should be added automatically. On the other hand, since you are setting here the global `CMAKE_CXX_FLAGS`, if you tried to do a _debug build_ (`./configure Debug`), I think you would end up with `... -O3 -DNDEBUG -O0 -g`, (or maybe the other way round). 

The release flags should be set in `CMAKE_CXX_FLAGS_RELEASE`, but again -- `-O3 -DNDEBUG` should be there automatically (if not building debug version). In short, **only the warning flags should be set here**. Other flags are handled automatically depending on the selected `CMAKE_BUILD_TYPE`.

There is one small catch: For GCC, CMake doesn't add `-O0` to debug builds, hence the explicit `set(CMAKE_CXX_FLAGS_DEBUG  "${CMAKE_CXX_FLAGS_DEBUG} -O0")`. I don't know if that's the case for Clang too (the verbose make or `message(STATUS "${CMAKE_CXX_FLAGS_DEBUG}")` is probably the easiest way to find out). See [here](https://cristianadam.eu/20190223/modifying-the-default-cmake-build-types/) for more info.

Reply:
Excellent!  I believe I've fixed these issues.  I've tried debug and release compiles with Clang and everything appears to be correct as far as compiler flags.  I'm still trying to navigate the CMake minefield... and I still don't like it!  :)